### PR TITLE
add 'update_tag' option

### LINKED ISSUE #171: Release time does not update
## expected behaviour

I expected that when i generate a release through GitHub-Actions as such:
```yaml
on:
  push:
    branches: [ main, release-publish ]
  schedule:
    - cron: "* */6 * * *"
[...]
    - name: Release
      uses: softprops/action-gh-release@v1
      with:
        tag_name: linux-latest
        files: |
          client/dist/client
```
that the release which is generated points to the commit which triggered the action and the time of the push.

## actual behaviour
The assets of the release are the correct after the workflow run, but the release points to the wrong commit and therefore states a wrong time.

## Issue in more detail

In [my project](https://github.com/emcruise/tor-rootkit) I have a cron job, which regularly updates my releases. Even though the assets change correctly the release states a wrong time and points to a wrong commit, which can be seen [here](https://github.com/emcruise/tor-rootkit/tree/c56ad74cb60eb6db5fca58468250d6800ef26ee7). The wrong time could result to me living in a UTC+1 time zone which could give my commits a wrong time, but this still doesn't explain why the release points to the wrong commit. When I delete the releases and regenerate them through GitHub-Actions they work correctly and point to the correct commit with the correct time.

### PR DESCRIPTION
**Pull request by iTrooz**
This PR  adds the 'update_tag' option

Fixes #171

### CODE REVIEW

**Comment by sithlord48** [comment]
I need this feature for continuous releases . @softprops  How can I help test this ?

**Comment by mkurz** [comment]
@softprops Can you please take a look here finally? This is a very useful feature and [our project](https://github.com/playframework/playframework) would highly benefit from it. Thank you very much!

**Comment by chenrui333** [comment]
@iTrooz can you fix the merge conflicts? Thanks!

**Comment by chenrui333** [comment]
@iTrooz ping

**Comment by iTrooz** [comment]
Hey, unless the maintainer shows interest in merging this, I'm not interested in updating it, sorry

**Comment by iTrooz** [comment]
Oops, I just realized that you *are* part of this repository. I will update this PR soon

**Comment by chenrui333** [comment]
yeah, please :)

**Comment by iTrooz** [comment]
Why did I think making a refactor on a feature PR was a good idea..

Anyway, done !

Note that I have no idea how github orders release, using `update_tag` may not bring the release to the top. But it *will* avoid a spam of notifications to users subscribed :)

**Comment by iTrooz** [comment]
@chenrui333 Ready for review/merge

I already tested this PR, but if anyone wants to double-check, please do so

File: src/github.ts [hunk 1]
```diff
@@ -297,6 +341,29 @@ export const release = async (
...
+        repo,
+        ref: "refs/tags/"+existingRelease.tag_name,
+        sha: newCommitSha
+      })
+
+      console.log(`Updated ref/tags/${existingRelease.tag_name} to ${newCommitSha}`);
+      
+      // give github the time to draft the release before updating it
+      // Else, I think we would have a race condition with github to update the release
+      await sleep(2000);
```
Comment (softprops) [inline src/github.ts:364]: There is something inherently unreliable and inefficient about [relying on time as a synchronization primitive](https://xeiaso.net/blog/nosleep/)

It’s possible that sleeping 2 seconds isn’t long enough resulting in a noneffective approach and it is also possible that is is way to long resulting in wasted gh action time.

Is it possible to restructure this without sleeping?

File: README.md [hunk 2]
```diff
@@ -194,6 +194,7 @@ The following are optional as `step.with` keys
...
 | `make_latest`              | String  | Specifies whether this release should be set as the latest release for the repository. Drafts and prereleases cannot be set as latest. Can be `true`, `false`, or `legacy`. Uses GitHub api defaults if not provided                                                                                                                                                                                                                            |
+| `update_tag`               | Boolean | Update the tag of the release to the current commit. This will also update the release time. Default is false                                                                                                                                                                                                                                                                                                                                   |
```
Comment (softprops) [inline README.md:197]: I have some reservations about this. Action gh release is an action for managing GitHub releases, the product feature

the concern is that with these changes the scope is increasing to also include managing git state something that feels possible with git itself external to this action. 

Would it be possible to do that as a separate workflow step or possibly trigger that then triggers the workflow that manages gh release data?

File: src/github.ts (hunk 1 above)
Comment (iTrooz) [reply src/github.ts:364]: I fully understand why sleeping is not a perfect solution.
That said, this is certainly the easiest option to implement.

Another solution that could work would be to make a request to the github API every second to check if the draft was done yet. Would you prefer this ?

**Comment by iTrooz** [comment]
Closing in favor of https://github.com/softprops/action-gh-release/issues/171#issuecomment-1229559873 (See also https://github.com/softprops/action-gh-release/pull/293#discussion_r1687263755)

**Pull request closed without merging**